src/App.tsx

```diff
@@ -1,93 +1,160 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+
+import React, { useEffect, useState, useRef } from 'react';
+import { Status, Todo } from './types';
+import { TodoList } from './components/TodoList';
+import { TodosFilter } from './components/TodosFilter';
+import { TodosContext } from './TodosContext';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [inputValue, setInputValue] = useState('');
+  const [activeFilter, setActiveFilter] = useState(Status.All);
+
+  const completedTodos = todos.filter(todo => todo.completed);
+  const areAllTodosCompleted = completedTodos.length === todos.length;
+  const isAnyTodoCompleted = completedTodos.length !== 0;
+  const notCompletedTodoCount = todos.length - completedTodos.length;
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (activeFilter) {
+      case Status.All:
+        return true;
+
+      case Status.Active:
+        return !todo.completed;
+
+      case Status.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  useEffect(() => {
+    inputRef.current?.focus();
+
+    setTodos(JSON.parse(localStorage.getItem('todos') || '[]'));
+    setActiveFilter(JSON.parse(
+      localStorage.getItem('activeFilter') || `"${Status.All}"`,
+    ));
```

Comment:
Parsing localStorage data directly can lead to potential issues if the data is corrupted or not in the expected format. Consider using a try-catch block when parsing JSON.